ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/util/SimpleOffsetSerializer.java

```diff
@@ -210,14 +212,19 @@ public static Bytes serializeVariableCompositeList(
   public static <T> T deserialize(Bytes bytes, Class<T> classInfo) {
     MutableInt bytePointer = new MutableInt(0);
     if (!isPrimitive(classInfo)) {
-      return SSZ.decode(
-          bytes,
-          reader -> {
-            final T result =
-                deserializeContainerErrorWrapper(classInfo, reader, bytePointer, bytes.size());
-            assertAllDataRead(reader);
-            return result;
-          });
+      Optional<ViewType> maybeViewType = ViewType.getType(classInfo);
+      if (maybeViewType.isPresent()) {
```

Comment:
Does this mean we can remove all of the tree-backed classes from `SimpleOffsetSerializer.setConstants()`?

Reply:
Seems there are `classLengthBounds` which are still actual. I was going to do the final cleanup in a subsequent PR while removing old SSZ code